Recent commit causes new "Attempting to deserialize object on a CUDA device" exception

### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.


### YOLOv5 Component

PyTorch Hub

### Bug

I've been using YOLOv5 via PyTorch Hub for a few months now, but #7878 broke the deployment. I went from reliably running inference to getting the following error:

```
[ERROR] Exception: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.. Cache may be out of date, try `force_reload=True` or see https://github.com/ultralytics/yolov5/issues/36 for help.
```

The referenced issue was not helpful for solving the issue. What worked was reverting back to commit 
eb1217f3ac93fa906761216b16ae1b9435cbb850, which is just before the recent OpenVINO and MPS merges that seem to have caused the issue.

I'm now running YOLOv5 at commit eb1217f3ac93fa906761216b16ae1b9435cbb850 using the PyTorch Hub local feature and everything's back to normal. I'm not too familiar with the workings of YOLOv5 so I'm not able to spot where in the recent commits the bug is.

### Environment

- YOLO: YOLOv5 🚀 2022-5-26 Python-3.8.13 torch-1.10.2+cu102 CPU
- OS: AWS Linux
- Python 3.8.13

### Minimal Reproducible Example

Running the following: 
```
model = torch.hub.load('ultralytics/yolov5', 'custom', path='./custom_weights.pt', force_reload=True, device='cpu')
```
Causes the following error, even though this has been working for months in production systems:

```
[ERROR] Exception: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.. Cache may be out of date, try `force_reload=True` or see https://github.com/ultralytics/yolov5/issues/36 for help.
```

To fix it, I changed this to:
```
model = torch.hub.load('./yolov5', 'custom', path='./custom_weights.pt', force_reload=True, device='cpu', source='local')
```
Where `yolov5` is a clone of `ultralytics/yolov5` at commit eb1217f3ac93fa906761216b16ae1b9435cbb850.


### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

--- a/models/experimental.py
+++ b/models/experimental.py
@@ -77,8 +77,8 @@ def attempt_load(weights, device=None, inplace=True, fuse=True):
     # Loads an ensemble of models weights=[a,b,c] or a single model weights=[a] or weights=a
     model = Ensemble()
     for w in weights if isinstance(weights, list) else [weights]:
-        ckpt = torch.load(attempt_download(w))
-        ckpt = (ckpt.get('ema') or ckpt['model']).to(device).float()  # FP32 model
+        ckpt = torch.load(attempt_download(w), map_location=device)
+        ckpt = (ckpt.get('ema') or ckpt['model']).float()  # FP32 model
         model.append(ckpt.fuse().eval() if fuse else ckpt.eval())  # fused or un-fused model in eval mode
 
     # Compatibility updates

